webroot/results/admin/persons_finish_unfinished.php

```diff
@@ -128,24 +128,58 @@ function showUnfinishedPersons () {
 
   #--- Walk over all persons from the Results table.
   $caseNr = 0;
+  $availableSpots = array(); // array of semiIds in progress
   foreach( $personsFromResultsWithoutId as $person ){
     list( $name, $countryId, $firstYear ) = $person;
     
     #--- Try to compute the semi-id.
-    $quarterId = removeUglyAccentsAndStuff( extractRomanName( $name ));
-    $quarterId = preg_replace( '/[^a-zA-Z ]/', '', $quarterId );
-    $quarterId = strtoupper( substr( preg_replace( '/(.*)\s(.*)/', '$2$1', $quarterId ), 0, 4 ));
-    if ( strlen ( $quarterId ) == 0 ) {
-      // if the name comes empty, invent a quarterId
-      $quarterId = 'XXXX';
-    } else if ( strlen( $quarterId ) < 4 ) {
-      // make sure the quarterId is 4-letter long
-      while ( strlen( $quarterId ) < 4 ) {
-        $quarterId .= $quarterId;
-      }
-      $quarterId = substr( $quarterId, 0, 4 );
+    $paddingLetter = 'U';
+    $neatName = strtoupper(preg_replace('/[^a-zA-Z ]/','',removeUglyAccentsAndStuff(extractRomanName($name))));
+    $nameParts = explode(' ',$neatName);
+    $lastName = $nameParts[count($nameParts)-1];
+    $restOfName = '';
+    for ($i=0;$i<count($nameParts)-1;$i++) {
+        $restOfName .= $nameParts[$i];
+    }
+    // follows a simple trick that prevents us from empty or too short restOfNames and provides the appropriate padding
```

Comment:
@jfly the trick not only guarantees that `$restOfName` is at least 4 characters long, but also provides the padding Us, just in case.  Those Us won't be used normally, though.
@Baiqiang your proposal is fine for me.